### ISSUE TITLE
check-requirements error is misleading when dependency is pinned in requirements-dev.txt and requirements.txt

### ISSUE THREAD

**Issue by firba1** [Yelp/requirements-tools#27]
# How to reproduce
1. Add a pinned requirement to requirements-dev.txt that's already in requirements.txt
2. run `check-requirements`
3. See error something like this:
```
E   AssertionError: Requirements are pinned in requirements-dev.txt but are not depended on in requirements-dev-minimal.txt!
E   
E   Usually this happens because you upgraded some other dependency, and now no longer require these.
E   If that's the case, you should remove these from requirements-dev.txt.
E   Otherwise, if you *do* need these packages, then add them to requirements-dev-minimal.txt.
E       - foo==3.4.1
1 failed, 3 passed in 3.01 seconds
```

In the situation where this _is_ depended on, this error is confusing.

**Comment by asottile** [comment]
Good catch! Probably the easiest fix is to add that scenario to the error message? Though I guess it could be directly detected. I wonder what happens if it "is" in requirements-dev-minimal.txt?

**Comment by firba1** [comment]
If it _is_ in requirements-dev-minimal.txt, you get the same error, actually!

I think adding messaging describing this scenario first makes the most sense, since I've run into that situation a lot more than either of the other two.

**Comment by asottile** [comment]
lol amazing. Yeah definitely a case that deserves a better message (at least there's _some_ error, Eh?)
